tools/dom-test-helpers.js

```diff
@@ -10,3 +11,18 @@ export function keyDown(element, keycode) {
 	event.code = keycode;
 	element.dispatchEvent(event);
 }
+
+export async function getUpdateCompleteAll(node) {
+
+	if (!node || (node.nodeType !== Node.ELEMENT_NODE && node.nodeType !== Node.DOCUMENT_NODE && node.nodeType !== Node.DOCUMENT_FRAGMENT_NODE)) {
+		throw new TypeError('Invalid node. Must be nodeType document, element or document fragment');
+	}
+
+	const getUpdateComplete = async elem => {
+		if (elem.updateComplete) await elem.updateComplete;
```

Comment:
This code purposefully waits for `updateComplete` before checking its children. My first implementation of this just collected all the promises in a flat array without waiting, but I was observing inconsistent results due to timing issues.